### DISCUSSION TITLE
Bisection is hard

### DISCUSSION THREAD

**Discussion by 64** [managarm/managarm#311] [General]
There are a few reasons why bisection is hard for us:
1. `master` branch is broken too often
2. Since we have tens of different repositories for different components, it's difficult to revert the state of just one repository (because it might rely on changes in e.g `libasync` just to compile)

As the project gets larger and more stable, it would be good to be able to bisect things. What can we do to make it easier?

**Comment by avdgrinten** [comment]
Two things come to mind:

- We could add the dependencies of the `managarm` repository as subprojects and modify the GitHub CI to build using these dependencies (instead of fetching packages from our package repository). However, that will not work for changes that affect both mlibc and managarm.
- To bisect changes over the entire set of packages, we could make xbbs export its `bootstrap-commits.yml` file that records the exact SHA1 hashes of the commits of *all* packages (including ports). We could then add a command to xbstrap to revert multiple packages to earlier commits.
